### DISCUSSION TITLE
I cannot re-enable 2FA authentication

### DISCUSSION THREAD

**Discussion by vietstone-ng** [community/community#50189] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi, 
I disable 2FA by accident.
I tried to re-enable it. I stuck at step 2. After downloading recovery codes, and press "I have saved my recovery codes", it shows the error message "Enabling two-factor authentication failed".
Please help.
Thank you.

**Comment by vietstone-ng** [comment]
Hello

**Comment by RONIE70** [comment]
If you know your password for GitHub.com but don't have the two-factor authentication credentials or your two-factor authentication recovery codes, you can have a one-time password sent to your verified email address to begin the verification process and regain access to your account.

Note: For security reasons, regaining access to your account by authenticating with a one-time password can take up to three business days. GitHub will not review additional requests submitted during this time.

You can use your two-factor authentication credentials or two-factor authentication recovery codes to regain access to your account anytime during the 3-5 day waiting period.

Type your username and password to prompt authentication.

Warning: If you protect your personal account with two-factor authentication but do not know your password, you will not be able to follow these steps to recover your account. GitHub can send a password reset email to a verified address associated with your account. For more information, see "[Updating your GitHub access credentials](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/updating-your-github-access-credentials#requesting-a-new-password)."

Under "Having problems?", click Use a recovery code or request a reset.

Screenshot of the 2FA page at sign in. A link with the text "Use a recovery code or request a reset" is outlined in orange.

Under "Locked out?", click Recover your account or unlink an email address.

Click I understand, get started to request a reset of your authentication settings.

Click Send one-time password to send a one-time password to all eligible addresses associated with your account. Only verified emails are eligible for account recovery. If you've restricted password resets to your primary and/or backup addresses, these addresses are the only addresses eligible for account recovery.

Under "One-time password", type the temporary password from the recovery email GitHub sent, then click Verify email address.

Choose an alternative verification factor.

If you've used your current device to log into this account before and would like to use the device for verification, click Verify with this device.
If you've previously set up an SSH key on this account and would like to use the SSH key for verification, click SSH key.
If you've previously set up a personal access token and would like to use the personal access token for verification, click Personal access token.
A member of GitHub Support will review your request and email you within three business days. If your request is approved, you'll receive a link to complete your account recovery process. If your request is denied, the email will include a way to contact support with any additional questions.

**Comment by airtower-luna** [comment]
If you get that message before you even try to set up a 2FA message (other than the recovery codes) it sounds like a bug. Try the bug report form to let support know: https://support.github.com/contact/bug-report

**Reply by vietstone-ng** [reply]
I write a ticket to Github Support 6 days ago, still waiting.

**Comment by papagunit** [comment]
Same issue, just tried enabling 2FA for the first time. Was able to receive the code via SMS, entered it, was provided the recovery codes and downloaded, but upon clicking "I have saved my recovery codes", I received the error "Enabling two-factor authentication failed".

![image](https://user-images.githubusercontent.com/8437025/226751981-7a6b3f45-2d8e-46c4-88e5-a453474c1c1b.png)

**Reply by vietstone-ng** [reply]
Did you manage to resolve the problem?

**Reply by papagunit** [reply]
@vietstone-ng nope. Just tried a moment ago, still the same result.

Are we supposed to see the codes in the grey box?

Perhaps it doesn't like my phone number carrier (GV)? Not sure why there should be any additional validation after "I have saved my recovery codes".

**Reply by chenyucheng0503** [reply]
Same issue, and also use GV. So I guess it's the problem

**Reply by ben-g123** [reply]
The recovery codes are not there, so they need to load. Also, did you click **download**?

**Comment by papagunit** [comment]
Just tried it today - completely different experience. Didn't ask for phone number, just gave a QR code for setting up with an authentication app. Got that set up and then it actually provided the recovery codes on the page (not just the blank grey box like I had before). All good!

**Reply by vincentditlevinz** [reply]
I encounter the same xp. I disabled 2 factors auth, re-enabled it and I am stucked at step 2.

**Comment by vietstone-ng** [comment]
I ended up talking to Github Support (https://support.github.com/tickets). Record a video to show them my problem. They understand it, then help me to fix quite fast.

**Comment by samantha-jp** [comment]
Same experience of seeing that "Enabling two-factor authentication failed" message. 
I clicked Cancel and confirm the action, it refreshed and showed the QR code to add in an authentication app. The rest is as described by papagunit.

**Reply by YuanTingHsieh** [reply]
This helps me, thanks!

**Comment by valyadvorzhak** [comment]
I tried 1000 times to do it on 2fa app Aegis and it always says " Two-factor code verification failed. Please try again. " Then I created a new application instance on the 2fa app Aegis (don't delete the old one), but used not a QR-code, but a text code, and entered this code on the Enable 2fa page on the Github. Then this new instance generated a new 6-digit code and I filled it in the GitHub input field. And it works! Hope it can help sombody!

**Reply by valyadvorzhak** [reply]
The problem was in the QR code or how my phone reads QR codes.
